nmstatectl/nmstatectl.py

```diff
@@ -98,6 +125,77 @@ def _filter_state(state, whitelist):
     return state
 
 
+def edit(args):
+    state = _filter_state(netinfo.show(), args.only)
+
+    if args.yaml:
+        suffix = '.yaml'
+        txtstate = pretty_yaml(state)
+    else:
+        suffix = '.json'
+        txtstate = pretty_json(state)
+    with tempfile.NamedTemporaryFile(suffix=suffix,
+                                     prefix='nmstate-') as statefile:
+        statefile.write(txtstate)
+        statefile.flush()
+
+        while True:
+            txtstate = run_editor(txtstate, suffix)
+            error = ""
+
+            if txtstate is None:
+                return 1
```

Comment:
You said the ctl is primarily for us, do we really need to distinguish these errors? Seems to me overkill at the moment since there is an API that could be used to get proper exceptions instead of calling nmstatectl and checking the returncode.

Reply:
changed